### ISSUE TITLE
Can i make kexecboot skip displaying the menu?

### ISSUE THREAD

**Issue by buzztiaan** [kexecboot/kexecboot#15]
And just start the only entry in my boot.cfg directly.

I tried a TIMEOUT=0 but that doesnt seem to do much.

**Comment by andrea-adami** [comment]
Hi,
the option is disabled by default.
You have to configure kexecboot with  --enable-timeout for it to work.

**Comment by buzztiaan** [comment]
But, does that allow me to do timeout=0 to skip displaying the menu?

**Comment by andrea-adami** [comment]
Yes,
once the binary is rebuilt with --enable-timeout you can set TMEOUT=0 in boot.cfg.
Otherwise it is ignored.

**Comment by buzztiaan** [comment]
the question was if that would then prevent showing the menu, but i guess so?

**Comment by andrea-adami** [comment]
Yes, it was designed to boot immediately without requiring user intervention.
You might also try to configure with --enable-fbui=no  to avoid any graphical output (if your machine is especially slow you could maybe have a 'flash' on screen, please try and report)

**Comment by jay7x** [comment]
I checked the code and found this: https://github.com/kexecboot/kexecboot/blob/master/src/kexecboot.c#L1084
I guess kexecboot cannot start w/o UI because of that check..

@buzztiaan you may want to comment this string to check with `--enable-fbui=no` (`--disable-fbui`?).
Another option is to use text UI which is faster so you may notice no 'blink'.

**Comment by andrea-adami** [comment]
Heh, I was suggesting to use text-ui alone, which is disabled by default.
One UI is necerssary OFC

To resume:
--enable-fbui=no
--enable-textui=yes

This is not widely used setup: the idea was to have a graphical menu.
Please let us know, thanks

**Comment by xsetiadi** [comment]
> Heh, I was suggesting to use text-ui alone, which is disabled by default.
> One UI is necerssary OFC
> 
> To resume:
> --enable-fbui=no
> --enable-textui=yes
> 
> This is not widely used setup: the idea was to have a graphical menu.
> Please let us know, thanks

I was able to build with this config. no more UI or kexecboot menu, but its takes 3 time reboot to the boot logo before it success to boot to the first entry os I have on my droid4

**Comment by buzztiaan** [comment]
i guess it helps that our used kernel doesnt even support any text console ;) 

currently building with: 
```
KEXECBOOT_CONF_OPTS = --enable-textui=yes \
        --enable-fbui=no \
        --enable-numkeys \
        --enable-timeout \
        --enable-debug
```

**Comment by xsetiadi** [comment]
> i guess it helps that our used kernel doesnt even support any text console ;)
> 
> currently building with:
> 
> ```
> KEXECBOOT_CONF_OPTS = --enable-textui=yes \
>         --enable-fbui=no \
>         --enable-numkeys \
>         --enable-timeout \
>         --enable-debug
> ```

did you build with it? and how is it going?

**Comment by jay7x** [comment]
> I was able to build with this config. no more UI or kexecboot menu, but its takes 3 time reboot to the boot logo before it success to boot to the first entry os I have on my droid4

Did it reboots 3 times by itself or you rebooted it 3 times to get it working?

**Comment by xsetiadi** [comment]
> > I was able to build with this config. no more UI or kexecboot menu, but its takes 3 time reboot to the boot logo before it success to boot to the first entry os I have on my droid4
> 
> Did it reboots 3 times by itself or you rebooted it 3 times to get it working?

reboot by itself. and the 4th boot it success. 

but now no more reboot when booting

just booting to the first entry os

**Comment by jay7x** [comment]
> i guess it helps that our used kernel doesnt even support any text console ;)

Not sure I tried to run kexecboot on kernel w/o text console..
@andrea-adami do you remember if we tried it?

Though it may work with `--disable-debug`.

Another option for you is to build w/o any UI. Need to comment one line in code I referred above and build with `--disable-fbui --disable-textui` (mostly just guessing so it's up to you to try).

**Comment by xsetiadi** [comment]
> > i guess it helps that our used kernel doesnt even support any text console ;)
> 
> Not sure I tried to run kexecboot on kernel w/o text console..
> @andrea-adami do you remember if we tried it?
> 
> Though it may work with `--disable-debug`.
> 
> Another option for you is to build w/o any UI. Need to comment one line in code I referred above and build with `--disable-fbui --disable-textui` (mostly just guessing so it's up to you to try).

ok I can try.

and what the function of this config --enable-numkeys

**Comment by jay7x** [comment]
> and what the function of this config --enable-numkeys

IIRC it allows you to choose item in menu by pressing corresponding number button (e.g. press 1 to choose 1st item in menu). If you have no hardware keyboard you don't need it I guess.

@xsetiadi @buzztiaan Are you guys working together on this? Just for my further understanding :)

**Comment by buzztiaan** [comment]
we are both involved in Maemo Leste which uses kexecboot on Motorola Droid 4 , xsetiadi is just mirroring my efforts a bit :)

**Comment by xsetiadi** [comment]
buzztiaan right. im just nothing, testing and find the bug. coding is not my speciality

**Comment by jay7x** [comment]
ah, nice! Didn't know kexecboot is used there :-D
So you're just using it to boot 1st kernel available in "unattended mode", right?

**Comment by andrea-adami** [comment]
I see you use it on droid4.

About the 3 reboots, we should put  Tony Lindgren <tony@atomide.com> in c/c because he helped us to add new configure options for that model.

**Comment by xsetiadi** [comment]
I often had this random reboot after chose os on kexecboot menu. but usually its just once.

and the next boot it's booting to the os.

but this problem only happen when first time installs kexecboot.
after a couple of times it just normal

**Comment by buzztiaan** [comment]
i have no clue what 'unattented mode' should mean , this linux image is booted first by the droid4 now ; https://github.com/tmlind/buildroot/tree/droid4-kexecboot-2017.11

thats indeed by @tmlind  ;) he's aware of my/our effort to get it without any display to autoboot the first image :)

**Comment by greguu** [comment]
to get no output on the screen during kexecboot,  the timeout option in
kexecboot does the trick to load your first image, but you may as well as
disable the framebuffer and any console output in the kernel options.

**Comment by jay7x** [comment]
Hello! What is the current status of the problem? Do you need any further help here? Can I close the issue?
